### ISSUE TITLE
MSYS usage

### ISSUE THREAD

**Issue by ericLemanissier** [copperspice/copperspice#176]
I just configured, built, and used copperspice with mingw, from a standard windows command prompt, without MSYS/MSYS2 installed on the machine.  If MSYS is needed, what is it for?  [ edited for content ]

**Comment by bgeller** [comment]
The following text is is part of our CS Overview documentation for building CS using Windows and MinGW.

_Windows does not provide a Bourne shell as a standard component. In order to build CopperSpice a shell program is required to run the configure scripts._

Other platforms like Unix and Mac provide a built in shell environment where programs like BASH are GREP are common to Unix users. Windows does not provide a compatible shell environment but thankfully there are some good ports in the open source community. MSys is actually a set of programs which includes MinGW and lots of other very useful Unix tools.  If you want to run gdb for debugging you will want MSys installed. It is also required to run several of the git tools. 

MSys has been around for many years it still works great. MSys2 is a bit newer but we found some of the utilities are not standard and we do not recommend using it. In addition, MSys2 is a bit harder to install and adds utilities we found to be unnecessary.

**Comment by ericLemanissier** [comment]
> Windows does not provide a Bourne shell as a standard component. In order to build CopperSpice a shell program is required to run the configure scripts.

What configure script are you talking about ? there don't seem to be any.


> MSys is actually a set of programs which includes MinGW and lots of other very useful Unix tools


mingw is certainly not provided by the msys version you recommend: this msys dates from 2013 and GCC 8 is 5 years younger



> If you want to run gdb for debugging you will want MSys installed

I don't think so. gdb is included in the mingw package you recommend on https://www.copperspice.com/documentation-mingw.html


In the end, the only usage of mingw left are "GREP" and "other very useful Unix tools". It may not be the best to ask users to download some really outdated software, if it is not needed in the first place.

**Comment by bgeller** [comment]
Answering a few additional questions from multiple users.

All of the comments in this issue are about building CS with MinGW and none of them pertain to MSVC. A CS team member asked us about two years ago to provide separate documentation for building with MinGW and building with MSVC. It was a good request since there are different requirements. 

As per our documentation, building CS involves two steps. This is the same for every platform. MSVC simply does both for you.

(1) Running CMake, which is the configuration step 
(2) Running Ninja, compile and link process
  
Both MSys and MinGW come from the same project. We currently build with MinGW 7.3 on Windows, this is the lowest version we support on any platform. It makes perfect sense for our CI to test on what we support. A full list of every supported platform and the lowest version we use in CI is documented. 

There are several builds of MinGW and it can be confusing. We have listed the supported compilers and provided a link to 7.3 and 8.1, a link to 9.2 will be added soon. We are currently using C++17 to build CopperSpice and these provide our objective.

Of course anyone can use what they feel is of value to their project. If you have a requirement to use something newer, then please do so. Also keep in mind that updating to bleeding edge does not work for many. Rest assured, if someone has a problem open an issue on github, document what you did, and we will add this to our road map. 

https://www.copperspice.com/documentation-mingw.html

In summary, the CopperSpice team recommends MinGW users on Windows install MSys. As we test a new version (not MSys2) documentation and our web site will be updated so users have more choices.

**Comment by ericLemanissier** [comment]
closing the issue as my suggestions are being edited and deleted without any explanation.
